lib/online_order.rb

```diff
@@ -0,0 +1,85 @@
+require 'csv'
+require_relative 'order'
+require_relative 'customer'
+
+ONLINE_FILE = 'support/online_orders.csv'
+
+module Grocery
+  class OnlineOrder < Order
+    attr_reader :status, :customer
+
+    def initialize(id, products, customer_id, status=:pending)
+      super(id, products)
+      @customer = Customer.find(customer_id)
+      @status = status
+    end
+
+    # if the online order instance has no products, returns the total provided by Order#total (no shipping)
+    # otherwise, returns total provided by Order#total + $10 for shipping
+    def total
+      super
+      if @products.empty?
+        return super
+      else
+        return super + 10
+      end
+    end
+
+    # returns false if the product name is already included in the online order's products,
+    # raises an ArgumentError if the online order status is processing, shipped or complete,
+    # otherwise adds product to the instance's product hash and returns true
+    def add_product(product_name, product_price)
+      if @products.keys.include?(product_name)
+        return false
+      elsif [:processing, :shipped, :complete].include?(@status)
+        raise ArgumentError.new("The order status is #{@status}. New products can no longer be added to this order.")
+        return false
+      else
+        @products[product_name] = product_price
```

Comment:
Since `raise`-ing an exception immediately bails out of the method, the `return` on line 36 will never be reached.